### PR TITLE
Adding documentation

### LINKED ISSUE #143: Write up and publish documentation on how tol-lab-share supports volume tracking
**User story**
As developers we need to write and publish proper documentation for `tol-lab-share`. The documentation can include the following:

- An architectural overview on `tol-lab-share`.
- A setting up guide along with instructions on how to set up the underlying infrastructure for `tol-lab-share` to act as a queue consumer.
- About how `tol-lab-share` supports the volume tracking process.

To refer the documentation process already set-up for `traction-service`, please use this story: https://github.com/sanger/traction-service/issues/1394

The documentation should be published in GitHub Pages (e.g. Volume Tracking documentation for [`traction-service`](https://sanger.github.io/traction-service/volume-tracking/))

**Who are the primary contacts for this story**
Dasun, Steve, Stuart

**Who is the nominated tester for UAT**

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Documentation should be published in https://sanger.github.io/tol-lab-share
- [ ] A section should exist in the documentation detailing how `tol-lab-share` supports Volume Tracking.
- [ ] A section should exist on the overall architecture on how `tol-lab-share` consumes messages from RabbitMQ broker(s).

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- https://github.com/sanger/traction-service/issues/1394

**Additional context**

### PR DESCRIPTION
**Pull request by dasunpubudumal**
Closes #143 

#### Changes proposed in this pull request

- Volume tracking documentation for `tol-lab-share`: https://sanger.github.io/tol-lab-share/
- API docs for `tol-lab-share`: https://sanger.github.io/tol-lab-share/api-docs/

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: documentation/docs/lab-share-lib.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# lab-share-lib
+
+`lab-share-lib` is a separate [library](https://github.com/sanger/lab-share-lib) that encapsulates AMQP connectivity logic, message deserialization and schema validation. Sections below are some discussions on how it encapsulates logic related to messaging infrastructure and schema validation.
+
+## Connectivity with Message Queues
+
+`tol-lab-share` establishes AMQP connections with a RabbitMQ broker. `lab-share-lib`, therefore, uses [`pika`](https://pika.readthedocs.io/en/stable/) library to manage connectivity logic and handle failures. `pika` uses several callback functions to handle messaging logic, and these are captured in [`lab_share_lib/rabbit/async_consumer.py`](https://github.com/sanger/lab-share-lib/blob/6724ec3c5053b75bb2a33958621930d9bb876a31/lab_share_lib/rabbit/async_consumer.py) in `lab-share-lib`.
+
+```python linenums="1" title="Snippets on connectivity logic and callbacks with RabbitMQ broker"
```
Comment (sdjmchattie) [inline documentation/docs/lab-share-lib.md:9]: I didn't know these were flags you could set on these code blocks.  Interesting.  Thanks.

File: documentation/docs/lab-share-lib.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# lab-share-lib
+
+`lab-share-lib` is a separate [library](https://github.com/sanger/lab-share-lib) that encapsulates AMQP connectivity logic, message deserialization and schema validation. Sections below are some discussions on how it encapsulates logic related to messaging infrastructure and schema validation.
```
Comment (sdjmchattie) [inline documentation/docs/lab-share-lib.md:3]: I noticed there are some cases where you've put in a new line after the end of a sentence, and some where you haven't.  Not sure if you want to take the time to tweak these.

**Review by sdjmchattie — APPROVED** [review]

**Review by harrietc52 — APPROVED** [review]
This is amazing!!!!!! Definitely 'Approve' 🥇 

Just a small note, the link to https://sanger.github.io/tol-lab-share/SETUP_DOCKER.md under https://sanger.github.io/tol-lab-share/#setting-up-with-docker doesn't seem to work for me. 

How did you generate all the docs? Did you use CoPilot or write it yourself?

Super nice, definitely worth rolling out. Thanks for doing this first time around (sorry just realised second, after Traction).

**Comment by dasunpubudumal** [comment]
> This is amazing!!!!!! Definitely 'Approve' 🥇
> 
> Just a small note, the link to https://sanger.github.io/tol-lab-share/SETUP_DOCKER.md under https://sanger.github.io/tol-lab-share/#setting-up-with-docker doesn't seem to work for me.
> 
> How did you generate all the docs? Did you use CoPilot or write it yourself?
> 
> Super nice, definitely worth rolling out. Thanks for doing this first time around (sorry just realised second, after Traction).

Good catch. I'll update it 😄 Thanks!

**Comment by dasunpubudumal** [comment]
> This is amazing!!!!!! Definitely 'Approve' 🥇
> 
> Just a small note, the link to https://sanger.github.io/tol-lab-share/SETUP_DOCKER.md under https://sanger.github.io/tol-lab-share/#setting-up-with-docker doesn't seem to work for me.
> 
> How did you generate all the docs? Did you use CoPilot or write it yourself?
> 
> Super nice, definitely worth rolling out. Thanks for doing this first time around (sorry just realised second, after Traction).

Oh as to how I generated the docs: well all the volume tracking stuff were written by me. I don't think CoPilot understands what it is 😄 You just write stuff in a markdown file, and compile it to a bunch of static content that GitHub Pages serve. I took some help from CoPilot to run the shell script that does compiling them in CI though. 

The setup guides were taken from the README.md.

**Pull request merged**